Fabrice-TIERCELIN/eslint-plugin-unicorn
# Prefer indexof over findindex (#5)

* Add `prefer-indexof` rule

Add `prefer-indexof` rule

* Should not be here

* Add tests

* Handle arrow functions with block

* `prevent-abbreviations`: Fix crash with TypeScript parser (#913)

* `error-message`: Check `AggregateError` too (#921)

* Fix `false` import styles seen in reported problem messages (#910)

Co-authored-by: fisker Cheung <lionkay@gmail.com>

* Update `ava`, remove index prefix for snapshots (#926)

* Revert "`error-message`: Check `AggregateError` too (#921)" (#925)

* Revert "`error-message`: Check `AggregateError` too (#921)"

This reverts commit 08ddc78a431109eca8b151e2a8df641277a89ae6.

* Update error-message.js

* `import-style`: Fix crash on `babel-eslint` parser (#927)

* Fix integration test (#914)

* Update dependencies (#928)

* Update `.snap` files (#929)

* Rename to `prefer-array-index-of`

* Rename to `prefer-array-index-of`

* Rename to `prefer-array-index-of`

* Rename to `prefer-array-index-of`

* Rename to `prefer-array-index-of`

* Refactor to use `selector`

* New implementation with selector

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Rewrite `error-message` rule (#922)

* `prevent-abbreviations`: Add `ignore` option (#891)

* Reduce the fix area

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Only for test

* It seems that yield is needed here

* Accept any passive expression

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Add more cases

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js

* Changing value

* Add `getBinaryExpressionSelector`

* Make binaryExpressionSelector stricter

* Simplify

* Style

* Add many more test

* Add failed case

* Use  `visualize` test

* Improve report location

* Test `fix` function

* Test nested

* Update test

* Add  `getFunctionSelector`

* Exclude  `generator` and `async` functions

* Reuse variable

* check `isParenthesized`

* Fix style

* `consistent-function-scoping`: Allow ignoring arrow functions (#932)

* Update `create-rule` script to allow choosing fixable "whitespace" (#934)

* Update prefer-array-index-of.md

* Add `prefer-date-now` rule (#935)

* `prefer-ternary`: Refactor to use ESLint recommended way to extend fix range (#857)

* Update prefer-array-index-of.js

* Update prefer-array-index-of.js.md

* Make the code crash

* Restore the code

* Update prefer-array-index-of.md

* Add  `empty-brace-spaces` rule (#933)

* Adds weekly scheduled smoke tests (#923)

* No function name

* Update

* Typo

* Fix snapshots

* Update snapshots

* Test parameter used in deeper scope

* A edge case

* Tell it the same way as the message id

* Add details

* Fix the valid case and separate in blocks

* Reuse the text of prefer-array-index-of.md

* const index = foo.indexOf('foo');

* Update docs

* Update prefer-array-index-of.md

* Fix conflict

* Fix conflict in readme

Co-authored-by: fisker Cheung <lionkay@gmail.com>
Co-authored-by: futpib <futpib@gmail.com>
Co-authored-by: Sindre Sorhus <sindresorhus@gmail.com>
Co-authored-by: Ari Perkkiö <ari.perkkio@gmail.com>

If you're adding a new rule, please follow [these steps](../docs/new-rule.md).
